src/sentry/incidents/logic.py

```diff
@@ -1452,3 +1453,49 @@ def translate_aggregate_field(aggregate, reverse=False):
                 if translated_field == column:
                     return aggregate.replace(column, field)
     return aggregate
+
+
+def alert_rule_data_requires_async_lookup(organization, user, data, return_input_channel_id=False):
+    try:
+        from sentry.incidents.endpoints.serializers import AlertRuleTriggerActionSerializer
+
+        mapped_ids = {}
+        for trigger in data["triggers"]:
+            for action in trigger["actions"]:
+                a_s = AlertRuleTriggerActionSerializer(
+                    context={
+                        "organization": organization,
+                        "access": SystemAccess(),
+                        "user": user,
+                        "use_async_lookup": True,
+                    },
+                    data=action,
+                )
+                if a_s.is_valid():
+                    if (
+                        a_s.validated_data["type"].value == AlertRuleTriggerAction.Type.SLACK.value
+                        and not a_s.validated_data["input_channel_id"]
+                    ):
+                        if not return_input_channel_id:
+                            return True
+                        elif not a_s.validated_data["target_identifier"] in mapped_ids:
+                            (
+                                mapped_ids[a_s.validated_data["target_identifier"]],
+                                _,
+                            ) = get_target_identifier_display_for_integration(
+                                a_s.validated_data["type"].value,
+                                a_s.validated_data["target_identifier"],
+                                organization,
+                                a_s.validated_data["integration"].id,
+                                use_async_lookup=True,
+                                input_channel_id=None,
+                            )
+    except KeyError:
+        # Maybe data is invalid, no triggers/actions.
+        # Just return False - we'll serialize it, get the validation error, and give it to the user.
+        pass
+
+    if return_input_channel_id:
+        return mapped_ids
+
+    return False
```

Comment:
@iProgramStuff Let's be consistent with the return type here. We should either return `None` or `{}` here.

Reply:
It returns True/False if `return_input_channel_id` is False.

It returns a dict (populated or empty) if `return_input_channel_id` is True.

Do you think I should change that?

Reply:
Also I think I'm going to rename this var to `return_channel_ids`